ufo2fontir/src/source.rs

```diff
@@ -478,35 +474,66 @@ fn files_identical(f1: &Path, f2: &Path) -> Result<bool, WorkError> {
     Ok(true)
 }
 
-fn font_infos(
+/// Creates a map from UFO directory name => fontinfo.
+///
+/// That is, source.filename => fontinfo.
+fn font_infos<'a>(
     designspace_dir: &Path,
-    designspace: &DesignSpaceDocument,
-) -> Result<Vec<plist::Dictionary>, WorkError> {
-    let mut results = Vec::new();
+    designspace: &'a DesignSpaceDocument,
+) -> Result<HashMap<&'a String, norad::FontInfo>, WorkError> {
+    let mut results = HashMap::new();
     for source in designspace.sources.iter() {
         let ufo_dir = designspace_dir.join(&source.filename);
-        if !ufo_dir.join("fontinfo.plist").exists() {
-            continue;
-        }
-        results.push(load_plist(&ufo_dir, "fontinfo.plist")?);
+        let data_request = norad::DataRequest::none();
+        let font = norad::Font::load_requested_data(&ufo_dir, data_request)
+            .map_err(|e| WorkError::ParseError(ufo_dir, format!("{e}")))?;
+        results.insert(&source.filename, font.font_info);
     }
     Ok(results)
 }
 
+fn names(font_info: &norad::FontInfo) -> HashMap<NameKey, String> {
+    let mut result = HashMap::new();
+
+    let mut insert_if_present = |maybe_value: &Option<String>, name_id: NameId| {
+        if let Some(value) = maybe_value.as_ref() {
+            // For now let's very optimistically emit unicode names only
+            // <https://learn.microsoft.com/en-us/typography/opentype/spec/name#platform-encoding-and-language-ids>
+            result.insert(
+                NameKey {
+                    platform_id: 0, // Unicode
+                    encoding_id: 5, // Unicode full repertoire
```

Comment:
you sure you actually meant '5' here?
I believe that's for unicode variation sequences cmap 14:
https://learn.microsoft.com/en-us/typography/opentype/spec/cmap#unicode-platform-platform-id--0

I don't even see that defined in the name table's tables: https://learn.microsoft.com/en-us/typography/opentype/spec/name#platform-specific-encoding-and-language-ids-unicode-platform-platform-id--0

Reply:
That's a typo, it should say 4. Also I'm deleting this to match the fontmake behavior :)